docs/website/packages/codegen/generate.md

```diff
@@ -0,0 +1,61 @@
+---
+sidebar_position: 1
+title: Generating
+description: codegen's generate function
+---
+
+The generate function loads FHIR structure definition files, and integrates them into typescript code, which is determined by templates that you prepare. These templates are in [EJS format](https://ejs.co/). The files that generate the definitions can be found at `packages/codegen/definitions/fhir`.
+
+## Basic Example
+
+### Prepare your template
+
+Prepare an EJS formatted template, or multiple templates, and store them in a folder called `/templates` in your project, here is a basic example of a template:
+
+```ejs
```

Comment:
We should illustrate the name of the file, to hint at the naming convention:


```suggestion
```ejs title="resource-names.ts.ejs"
```